include/cuda/std/detail/__atomic

```diff
@@ -185,130 +192,216 @@ __host__ __device__ auto __cxx_atomic_alignment_unwrap(_Tp __value)
     return __cxx_atomic_alignment_unwrap(__value, integral_constant<bool, _LIBCUDACXX_ALIGNOF(_Tp) == sizeof(_Tp)>{});
 }
 
+template <typename _Tp>
+__host__ __device__ constexpr ptrdiff_t __cxx_atomic_skip(ptrdiff_t __value) {
+    using __t = decltype(*__cxx_atomic_alignment_unwrap(_Tp{}, integral_constant<bool, _LIBCUDACXX_ALIGNOF(_Tp) == sizeof(_Tp)>{}));
+    constexpr size_t __skip = sizeof(__t);
+    return __skip * __value;
+}
+
+template <class _Tp>
+using __atomic_strip_cv_t = typename cuda::std::remove_cv<_Tp>::type;
+
+template <class _Tp, typename _Tag>
+__host__ __device__ inline void __raw_atomic_store(_Tp* __a, __atomic_strip_cv_t<_Tp> __val, int __order, _Tag) {
```

Comment:
Flip the name parts maybe? I.e. `__atomic_raw_store`? This is not a strong opinion, but I _think_ I'd prefer that (OTOH I do recognize that the backend goes `__cxx_atomic` and not `__atomic_cxx` on names, so really either is fine I think - especially since this is further muddled by the CUDA backend having `_cuda` and the **very end**.)